src/ops/project.rs

```diff
@@ -1,23 +1,82 @@
+use nom_sql::ArithmeticOperator;
+use std::fmt;
 use std::collections::HashMap;
 
 use flow::prelude::*;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ProjectExpressionBase {
+    Column(usize),
+    Literal(DataType),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProjectExpression {
+    op: ArithmeticOperator,
+    left: ProjectExpressionBase,
+    right: ProjectExpressionBase,
+}
+
+impl ProjectExpression {
+    pub fn new(
+        op: ArithmeticOperator,
+        left: ProjectExpressionBase,
+        right: ProjectExpressionBase,
+    ) -> ProjectExpression {
+        ProjectExpression {
+            op: op,
+            left: left,
+            right: right,
+        }
+    }
+}
+
+impl fmt::Display for ProjectExpressionBase {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ProjectExpressionBase::Column(u) => write!(f, "{}", u),
+            ProjectExpressionBase::Literal(ref l) => write!(f, "(lit: {})", l),
```

Comment:
Should this just be the value? If I parse this right, we would currently end up with funny-looking output like `salary * (lit: 100)` rather than `salary * 100`.

Reply:
I added the `lit: ` since it outputs the emit index, and not the actual column name. I thought `0 * 100` might look slightly confusing.

https://github.com/ekmartin/distributary/blob/436333cce567b3c50e253fa44ba9f621c23a576a/src/ops/project.rs#L300

Reply:
Ah, fair point. I keep forgetting that the low-level operators only know about column indices...